modules/targets/darwin/default.nix

```diff
@@ -0,0 +1,74 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  inherit (pkgs.stdenv.hostPlatform) isDarwin;
+  inherit (lib.strings) escapeShellArg;
+
+  cfg = config.targets.darwin;
+
+  toDefaultsFile = domain: attrs:
+    pkgs.writeText "${domain}.plist"
+    (lib.generators.toPlist { } (filterAttrs (n: v: v != null) attrs));
+
+  toActivationCmd = domain: attrs:
+    "$DRY_RUN_CMD defaults import ${escapeShellArg domain} ${
+      toDefaultsFile domain attrs
+    }";
+
+  activationCmds = mapAttrsToList toActivationCmd cfg.defaults;
+in {
+  imports = [ ./keybindings.nix ./linkapps.nix ./search.nix ];
+
+  options.targets.darwin = {
+    defaults = mkOption {
+      type = types.submoduleWith { modules = cfg.defaultsSchema; };
```

Comment:
Instead of defining submodules, you could use a freeform module, see https://nixos.org/manual/nixos/stable/#sec-freeform-modules

Reply:
I've made it a freeform submodule so that we can add descriptions and types to known options.